### DISCUSSION TITLE
Unexpected error checking schema: TypeError: Failed to fetch

### DISCUSSION THREAD

**Discussion by syahrulfadli** [tinacms/tinacms#4229] [Help] [answered]
Hi,

Problem occur on production. I'm using Netlify to deploy the site.

I'm successfully login then after login window close immediately prompt error appear 
`Unexpected error checking schema: TypeError: Failed to fetch`. This error make me can't open my Collection.

![image](https://github.com/tinacms/tinacms/assets/64543548/519731ae-f972-4f9f-8527-e807e660fd58)

Then i check the the console heres what happened:
![image](https://github.com/tinacms/tinacms/assets/64543548/a7ec135f-198e-4d47-b749-1aa38c883849)

Addtional info may help:
- Site build with Hugo
- My build command (on netlify):
```
tinacms build && pnpm build
```
- My Tina `config.js`
```
export default defineConfig({
  branch: "main",
  clientId: process.env.CLIENT_ID || '', // Get this from tina.io
  token: process.env.TOKEN || '', // Get this from tina.io

  build: {
    outputFolder: "admin",
    publicFolder: "static",
  },
  media: {
    tina: {
      mediaRoot: "",
      publicFolder: "static",
    },
  },
  schema: {
    collections: [
      {
        name: "docs",
        label: "Docs",
        path: "content/docs",
        fields: [
          {
            type: "string",
            name: "title",
            label: "Title",
            isTitle: true,
            required: true,
          },
          {
            type: "rich-text",
            name: "body",
            label: "Body",
            isBody: true,
          },
        ],
      },
    ],
  },
  search: {
    tina: {
      indexerToken: process.env.INDEX_TOKEN || '',
      stopwordLanguages: ['eng']
    },
    indexBatchSize: 100,
    maxSearchIndexFieldLength: 100
  }
});
```

**Comment by kldavis4** [comment]
Your CSP is blocking access to cognito

**Reply by syahrulfadli** [reply]
Thank you for the answer. 

I try to solve the problem by injecting snippet meta tag with new CSP policy to allow it (using netlify feature) but it's still blocking the js request. The temporary solution I use is by using google chrome extension to rewrite CSP policy.

Do you have any references about this case?

**Reply by kldavis4** [reply, marked as answer]
You need to remove the CSP directives that are blocking the admin client from accessing aws cognito. I don't believe netlify has any default CSP but I can't advise on the specifics - I suggest you follow-up on one of their support channels if you aren't able to resolve it yourself.

**Reply by syahrulfadli** [reply]
Okay I found it. It's from my `netlify.toml`. I'm recently using Hyas to build a web they put netlify config file on starter folder project when you create new project with Hyas.

Here the snippet.
```
[[headers]]
  for = "/*"
  [headers.values]
    Strict-Transport-Security = "max-age=31536000; includeSubDomains; preload"
    X-Content-Type-Options = "nosniff"
    X-XSS-Protection = "1; mode=block"
    Content-Security-Policy = "default-src 'self'; manifest-src 'self'; connect-src 'self'; font-src 'self'; img-src 'self' https://avatars.githubusercontent.com data:; script-src 'self' 'nonce-dXNlcj0iaGVsbG8iLGRvbWFpbj0iaGVua3ZlcmxpbmRlLmNvbSIsZG9jdW1lbnQud3JpdGUodXNlcisiQCIrZG9tYWluKTs=' 'sha256-aWZ3y/RxbBYKHXH0z8+8ljrHG1mSBvyzSfxSMjBSaXk='; style-src 'self'"
    X-Frame-Options = "SAMEORIGIN"
    Referrer-Policy = "strict-origin"
    Permissions-Policy = "geolocation=(self), microphone=(), camera=()"
    Cache-Control= '''
    public,
    max-age=31536000'''
    Access-Control-Allow-Origin = "*"
```
 Thanks in advance @kldavis4  :+1:
